### PR TITLE
feat(tabstops-auto-impl): Add tab stops requirement evaluator

### PR DESCRIPTION
**Pull request by sfoslund**
#### Details

Add `TabStopsRequirementEvaluator` and unit tests. This class will be leveraged by automated tab stops analyzer added in future PRs. 

##### Motivation

Support future PRs for this feature.

##### Context

A subsequent PR will add logic to call into this evaluator and use it to generate tab stops results automatically. 

#### Pull request checklist

- [n/a] Addresses an existing issue: #0000
- [x] Ran `yarn fastpass`
- [x] Added/updated relevant unit test(s) (and ran `yarn test`)
- [x] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] PR title *AND* final merge commit title both start with a semantic tag (`fix:`, `chore:`, `feat(feature-name):`, `refactor:`). See `CONTRIBUTING.md`.
- [n/a] (UI changes only) Added screenshots/GIFs to description above
- [n/a] (UI changes only) Verified usability with NVDA/JAWS

### CODE REVIEW

File: src/tests/unit/tests/injected/tab-stops-requirement-evaluator.test.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        ]);
+
+        const correctTabStops = new Set<HTMLElement>([tabStopElement1, tabStopElement2]);
+        expect(testSubject.getKeyboardNavigationResults(tabbableTabStops, correctTabStops)).toEqual(
+            [],
+        );
+    });
+
+    test('addFocusOrderResults', () => {
+        htmlElementUtilsMock
```
Comment (waabid) [inline src/tests/unit/tests/injected/tab-stops-requirement-evaluator.test.ts:51]: We like to break up tests by cases, so in this case the null and non-null scenarios will be separate.

File: src/tests/unit/tests/injected/tab-stops-requirement-evaluator.test.ts [hunk 2]
```diff
@@ -0,0 +1,102 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (waabid) [inline src/tests/unit/tests/injected/tab-stops-requirement-evaluator.test.ts:1]: Tests look good but should be broken by case to be consistent our usual test style.

**Review by waabid — APPROVED** [review]
LGTM! Would recommend breaking the test cases to separate `test()` sections before merging.

File: src/injected/tab-stops-requirement-evaluator.ts [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+export interface TabStopsRequirementEvaluator {
+    getKeyboardNavigationResults(
+        tabbableTabStops: FocusableElement[],
+        actualTabStops: Set<HTMLElement>,
+    ): TabStopRequirementResult[];
+    getFocusOrderResult(
+        lastTabStop: HTMLElement,
+        currentTabStop: HTMLElement,
+    ): TabStopRequirementResult;
+    getTabbableFocusOrderResults(tabbableTabStops: HTMLElement[]): TabStopRequirementResult[];
```
Comment (waabid) [inline src/injected/tab-stops-requirement-evaluator.ts:28]: This should be `FocusableElement[]` to be consistent.

Comment (sfoslund) [reply src/injected/tab-stops-requirement-evaluator.ts]: I've fixed these issues in the latest commit, resolving. Thanks for the feedback!

**Pull request merged**